pkg/detectors/parseur/parseur.go

```diff
@@ -30,6 +33,14 @@ func (s Scanner) Keywords() []string {
 	return []string{"parseur"}
 }
 
+func (s Scanner) ScanChunk(chunk sources.Chunk) bool {
+	// TODO: Can |chunk.SourceMetadata| be nil?
+	if m, ok := chunk.SourceMetadata.GetData().(sources.GitSourceMetadata); ok {
```

Comment:
Just tested this. Sadly, it does not work — life is not so easy.

Reply:
Do you mean that you need to explicitly attempt to unmarshal the metadata to determine whether it's from Git?

Reply:
I thought that `GetData()` returned a top-level object, however, it actually returns an object like `{"bitbucket":{...}} | {"github":{...}} | {"gitlab":{...}}`. Only calling `GetX()` returns the inner object.

We'd need to construct an ugly switch statement to handle each SourceType, similar to: https://github.com/trufflesecurity/trufflehog/blob/b03cc302630ead9afbce8c16c74bf2c29123c0c1/pkg/engine/engine.go#L605-L609.